### ISSUE TITLE
Add build tool plugins via unjs/unplugin

### ISSUE THREAD

**Issue by jasikpark** [markerikson/react-prod-sourcemaps#1]
It would be great to have a vite plugin, among other build tools, to enable shipping this change w/ minimal futzing on the user's part.

## Refs: 
- https://vitejs.dev/guide/api-plugin.html
- https://rollupjs.org/plugin-development/
- https://rollupjs.org/plugin-development/#this-getcombinedsourcemap

probably just needs a post-build hook and a grep of the build directory?

**Comment by markerikson** [comment]
Yeah, someone specifically suggested using Unplugin:

- https://twitter.com/imabhiprasad/status/1697290374660710545

**Comment by jasikpark** [comment]
also cc @JonasBa https://github.com/facebook/react/pull/26446#issuecomment-1701408043

**Comment by jasikpark** [comment]
can start w/ https://github.com/unplugin/unplugin-starter

**Comment by jasikpark** [comment]
Fixed by #4, though I'm running into an issue:

```
> nebula-web@0.0.0 build /Users/calebjasik/Git/defined.net/webclient
> vite build

vite v4.4.4 building for production...
✓ 2614 modules transformed.
computing gzip size (0)...[react-sourcemaps] Cannot find assets/server-ac67876d.js.map
✓ built in 8.33s
error during build:
Error: Cannot find assets/server-ac67876d.js.map
    at loadSourcemap (/Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/@acemarke+react-prod-sourcemaps@0.0.4/node_modules/@acemarke/react-prod-sourcemaps/lib/index.js:63:15)
    at rewireSourceMapsFromGeneratedAssetList (/Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/@acemarke+react-prod-sourcemaps@0.0.4/node_modules/@acemarke/react-prod-sourcemaps/lib/unplugin.js:44:105)
    at Object.writeBundle (/Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/@acemarke+react-prod-sourcemaps@0.0.4/node_modules/@acemarke/react-prod-sourcemaps/lib/unplugin.js:80:17)
    at file:///Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/rollup@3.26.2/node_modules/rollup/dist/es/shared/node-entry.js:25332:40
    at async Promise.all (index 0)
    at async PluginDriver.hookParallel (file:///Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/rollup@3.26.2/node_modules/rollup/dist/es/shared/node-entry.js:25260:9)
    at async file:///Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/rollup@3.26.2/node_modules/rollup/dist/es/shared/node-entry.js:26572:13
    at async catchUnfinishedHookActions (file:///Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/rollup@3.26.2/node_modules/rollup/dist/es/shared/node-entry.js:25698:16)
    at async build (file:///Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/vite@4.4.4_@types+node@18.15.13/node_modules/vite/dist/node/chunks/dep-abb4f102.js:47923:22)
    at async CAC.<anonymous> (file:///Users/calebjasik/Git/defined.net/webclient/node_modules/.pnpm/vite@4.4.4_@types+node@18.15.13/node_modules/vite/dist/node/cli.js:822:9)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by markerikson** [comment]
Interesting. I wonder if the assets have actually been written to disk at that point?

paging @JonasBa ...

**Comment by JonasBa** [comment]
@jasikpark this could be a path issue as well. Seems like we look for the path in assets, but we might require some similar logic to webpack where we join the path based on output dir.

```
webpack(compiler) {
        ....
        const outputPath = compilation.outputOptions.path ?? path.resolve();
        const assets = Object.keys(compilation.assets).map(asset => path.join(outputPath, asset));
        ....
    }
```
but for vite, we just do

```
writeBundle(_outputOptions, bundle) {
       rewireSourceMapsFromGeneratedAssetList(Object.keys(bundle), pluginOptions);
 },
```
Can you verify that /assets/server-ac67876d.js.map should not in fact be /build_dir/assets/server-ac67876d.js.map where build dir is your output path?

**Comment by jasikpark** [comment]
> Can you verify that /assets/server-ac67876d.js.map should not in fact be /build_dir/assets/server-ac67876d.js.map where build dir is your output path?

Nope, they are there - each source bundle has a `.js.map` in a separate file next to it...

**Comment by JonasBa** [comment]
@jasikpark I'll attempt to recreate it and see what's happening. Was a busy Monday so didn't have much time to look into it, but I made progress on adding some tests, so I need to open a PR and look into this, hopefully, the test suite catches is. Meanwhile if you have a repo or a branch somewhere that I can use to test it would be greatly appreciated!

**Comment by jasikpark** [comment]
I'll see if I can create a public repro for yah 👍

**Comment by JonasBa** [comment]
@jasikpark no need for the repro, see https://github.com/markerikson/react-prod-sourcemaps/pull/5

**Comment by JonasBa** [comment]
@jasikpark can you test your build with the latest release of the plugin?

**Comment by markerikson** [comment]
Just published v0.0.5:

https://github.com/markerikson/react-prod-sourcemaps/releases/tag/0.0.5

**Comment by jasikpark** [comment]
closing this! i'll continue bothering about things in #9 till I get setup right :p
